feat: simplify archived post indicator

jamesleesaunders opened the pull request:
Fixes https://github.com/bluesky-social/social-app/issues/6707

## Proposal
This PR has been raised as a follow up for https://github.com/bluesky-social/social-app/pull/6156

Whilst I appreciate the potential necessity to indicate whether a post has been backdated in the past, when the above feature was added this did then conflict with the way I personally use Bluesky (and previously X/Twitter) as a microblog diary of my family life etc. Having bulk loaded all my old Twitter (and Facebook) history into Bluesky I was keen to maintain the original dates and scroll back chronologically to read them.

This is a proposal for a kind of half-way solution where, rather than having two dates appear on a post, we only show one (the 'createdAt' date) but make it more obvious (i.e red) that is is not the original 'indexedAt' date. Not only does this mean the post looks less cluttered, but I think it is more obvious.

## Changes
* Consolidate Indexed and Created date fields into one.
* Make Backdated indicator Red rather than Orange (more obvious?).
* Swap dates over on Feed also so can scroll back thought time.

## Example screenshots:
### Current Post:
<img width="606" alt="Screenshot 2025-01-16 at 22 50 26" src="https://github.com/user-attachments/assets/330a5465-c291-497f-afc4-696a11851a3a" />

### Proposed Post (consolidate down to 1 date):
<img width="600" alt="Screenshot 2025-01-17 at 18 06 55" src="https://github.com/user-attachments/assets/a3430909-13c8-4b32-bef4-d08f5e99cf2b" />

### Archived Message (remains the same):
<img width="407" alt="Screenshot 2025-01-16 at 22 53 31" src="https://github.com/user-attachments/assets/00b3ea08-9b8e-402b-8508-d4233dff7a37" />

### Non-Backdated Post (remains the same):
<img width="611" alt="Screenshot 2025-01-16 at 21 48 27" src="https://github.com/user-attachments/assets/8740e7cd-5f40-4162-bc8a-5b413e24b1e7" />

### Current Feed (shows indexed_at date):
<img width="613" alt="Screenshot 2025-01-17 at 18 11 12" src="https://github.com/user-attachments/assets/e0bf8122-3051-4615-9acc-4076f96d7e3e" />

### Proposed Feed (shows created_at date): 
<img width="612" alt="Screenshot 2025-01-16 at 22 38 49" src="https://github.com/user-attachments/assets/628f98ef-9d7b-492f-88a6-b8212d36123f" />

I appreciate this is somewhat undoing the previous feature, but I just wanted to raise this PR if only for discussion as to the impact to my diary kind of use case for Bluesky. See [my profile](https://bsky.app/profile/jamessaunders.bsky.social) for example where you will see my diary (imported from Twitter) going back to 2003 by all posts at the moment now mostly all say '2mo' (2 months).

jamesleesaunders committed:
Merge indexedAt and createdAt date indicators.

jamesleesaunders committed:
Add use created at on PostFeedItem.

jamesleesaunders commented:
Also see discussion on https://github.com/bluesky-social/social-app/issues/6707

jamesleesaunders committed:
Rename 'Backdated' to 'Archived'.

jamesleesaunders committed:
Merge branch 'main' into backdated_post_indicator

supergenius88 commented:
I concur that this update to the poor current experience has been highly sought after by those of us who had our Twitter history migrated over. It does an excellent job of maintaining a visual cue that a record is imported but corrects the date indexing / sorting problem that the current backdate implementation introduced. Please consider this a strong plus vote.

jamesleesaunders committed:
Merge branch 'main' into backdated_post_indicator

jamesleesaunders commented:
@estrattonbailey @mozzius Apologies for reaching out directly but would love your consideration for getting this PR merged?
As a back story there is also some discussion on Bluesky here https://bsky.app/profile/jamessaunders.bsky.social/post/3lhyeyp4kvs2y

qtragabriex commented:
Approve. Save the world.
![praise](https://github.com/user-attachments/assets/e513507b-7d60-49b8-a237-18a2f5ffc807)

pfrazee commented:
I definitely appreciate where yall are coming from, but I do think this change would increase misinformation risk. Many users do not read very closely, and they will react to what shows in the feed (or most visibly on a post) before fully understanding what's happening.

@jamesleesaunders your goal is to be able to scroll back through a feed and see the original timestamp. What about including the archival notice in the feed view, so that original import time is visible?

ds84182 commented:
> What about including the archival notice in the feed view, so that original import time is visible?

What about the inverse of this: Show the archival note in the feed view as "Archived at <index time>"? Then the post timestamp can be dedicated to the creation time.

Lucifer-is-my-pet commented:
there's a similar suggestion https://github.com/bluesky-social/social-app/issues/6707 which addresses the problem with imported vs originally posted dates, and I think it's the way it should work

surfdude29 commented:
@pfrazee @mozzius Has something changed to cause the archived post indicator to show up for some backdated posts but not for others? I don't see an indicator on any of these:

![IMG_1842](https://github.com/user-attachments/assets/f180f4f6-89ad-44ae-bddc-fef57a69bae0)

https://bsky.app/profile/t4t.bsky.social/post/3l76lqjk3b62o

and

![IMG_1843](https://github.com/user-attachments/assets/d9d5a9f7-2b07-48ee-858f-2ca3cb7b4a85)

https://bsky.app/profile/retr0.id/post/3k5l4vko6jp2p

and, ironically, the post used in the example screenshots in #6156:

![IMG_1844](https://github.com/user-attachments/assets/9a8a82c8-a29e-4d17-8977-9616aafad345)

https://bsky.app/profile/durial321.bsky.social/post/3l7bq5kibp42s

On this PR, @ds84182's suggestion comes closest to what I think the ideal solution would be:

> What about the inverse of this: Show the archival note in the feed view as "Archived at <index time>"? Then the post timestamp can be dedicated to the creation time.

However, I think the text note should refer to the `indexed_at` timestamp. So posts in feeds would be sorted by their claimed, created date, but there would be a clear, unmissable message displayed as part of the post in the feed that states:

`Archived post first seen by Bluesky on <date>`

The same archival note would be shown when looking at a single post and would also be shown when a backdated post is quoted.

I think these changes would strike a balance between fulfilling the goal of allowing sorting and browsing posts by creation date on one hand, while at the same time eliminating any misinformation risk, because the `indexed_at` timestamp would always be shown and clearly visible.

jamesleesaunders committed:
Add backdated icon to feed.

jamesleesaunders committed:
Merge branch 'main' into backdated_post_indicator

jamesleesaunders committed:
fix: add backdated icon to feed

jamesleesaunders commented:
> Has something changed to cause the archived post indicator to show up for some backdated posts but not for others? I don't see an indicator on any of these:
> https://bsky.app/profile/t4t.bsky.social/post/3l76lqjk3b62o
> https://bsky.app/profile/retr0.id/post/3k5l4vko6jp2p
> https://bsky.app/profile/durial321.bsky.social/post/3l7bq5kibp42s

These are indeed weird? They do indeed look 'backdated'  - but yet they are not showing the current archived icon or message.

jamesleesaunders commented:
I have taken a few of the comments above onboard and for now simply added the archived icon to the feed view also:
<img width="611" alt="Screenshot 2025-04-21 at 15 06 06" src="https://github.com/user-attachments/assets/30b278ab-00c0-46e1-b0a3-119b559b57b9" />

<img width="609" alt="Screenshot 2025-04-21 at 15 05 17" src="https://github.com/user-attachments/assets/1727dc18-2e15-4279-9b5f-f9064da42550" />
<img width="363" alt="Screenshot 2025-04-21 at 15 06 30" src="https://github.com/user-attachments/assets/89cb1b7b-77bd-47bf-bdd1-ed4054aa36f6" />

This, I think, at least addresses concerns that in feed view users can not easily tell the difference easing the misinformation risk - I am not sure if this now goes far enough.

I did experiment with the idea of putting both dates with "Archived post first seen by Bluesky on <date>" on the feed but this then clutters the feed with lots of text (and I think we need to keep the feed view as simple as possible).

jamesleesaunders committed:
feat: add archive notice dialog to feed view

jamesleesaunders committed:
fix: correct some type errors

jamesleesaunders commented:
> @jamesleesaunders your goal is to be able to scroll back through a feed and see the original timestamp. What about including the archival notice in the feed view, so that original import time is visible?

@pfrazee Thanks for your suggestion. I have implemented what I think has been suggested by also adding the archival notice dialog (and calendar icon) to the feed view.

This, I think, satisfies the various requirements expressed above and also from the first MR https://github.com/bluesky-social/social-app/pull/6156:

Terminology for readers:
* createdAt - Original 'Archive' date which the post claims to be from, original date from Twitter etc.
* indexedAt - Date when first seen on Bluesky network (the date the post was created or imported to Bluesky)
* Archival Notice - Popup dialog box explaining difference between the above two dates.
* Calendar Icon - A small warning icon highlighting to the user the post is likely imported.

Changes:
  * Posts are now ordered by createdAt date (rather than indexedAt date) in the feed.
  * Posts which have a difference between createdAt and indexedAt of greater than 1-day will display the warning calendar icon.
  * Clicking on the icon will display the Archival Notice on Feed and single Post views.
 
Requirements Met:
* Diary Users: Improved history in feed for users who want to look back through the past of a user and see posts in order including those imported from other social networks.
* Misinformation Risk: Improved indication to users when a post has been imported with a red warning icon in both feed and post views.
* Less clutter: Removed two dates on Post view making it clearer and easier to read the dates.

Video demonstrating changes:
https://github.com/user-attachments/assets/956f6533-77bc-46be-91eb-8efea8688a5a

jamesleesaunders commented:
@pfrazee @mozzius Are we on the right track with this one now?
Is it good to merge?

jamesleesaunders committed:
Merge branch 'main' into backdated_post_indicator

jamesleesaunders commented:
I have not resolved the merge conflicts and re-tested if anyone is interested in merging this in?
 @pfrazee @mozzius

hexadecimal233 commented:
For 2 years this still hasn't been merged 💀
